src/index.ts

```diff
@@ -0,0 +1,436 @@
+import {
+    DerEncodedPublicKey,
+    HttpAgentRequest,
+    PublicKey as PublicKeyInterface,
+    requestIdOf,
+    Signature,
+    SignIdentity,
+} from '@dfinity/agent';
+import {DelegationChain} from '@dfinity/identity';
+import {Principal} from '@dfinity/principal';
+import {Buffer} from 'buffer';
+
+window.Buffer = Buffer;
+const domainSeparator = Buffer.from(new TextEncoder().encode('\x0Aic-request'));
+let globalStoicOrigin = 'https://www.stoicwallet.com';
+
+class PublicKey implements PublicKeyInterface {
+    constructor(private readonly _der: Uint8Array, private readonly _type: string) {}
+
+    public getType() {
+        return this._type;
+    }
+    public toDer(): DerEncodedPublicKey {
+        return Object.assign(this._der, {__derEncodedPublicKey__: undefined});
+    }
+}
+
+export class StoicIdentity extends SignIdentity {
+    constructor(
+        protected override readonly _principal: Principal,
+        private readonly _publicKey: PublicKey,
+    ) {
+        super();
+    }
+
+    static disconnect() {
+        return stoicLogout();
+    }
+
+    /**
+     * @param {string} host Override the wallet host to connect to. This will be preserved across
```

Comment:
Added some JSDoc comments. I don't understand the other methods or I'd add comments to them as well.